src/iOS.Core/Services/BiometricService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Bit.Core.Abstractions;
+using Foundation;
+using LocalAuthentication;
+
+namespace Bit.iOS.Core.Services
+{
+    public class BiometricService : IBiometricService
+    {
+        private IStorageService _storageService;
+
+        public BiometricService(IStorageService storageService)
+        {
+            _storageService = storageService;
+        }
+
+        public bool SetupBiometric()
+        {
+            var state = getState();
+            return _storageService.SaveAsync("biometricState", state).IsCompletedSuccessfully;
+        }
+
+        public bool ValidateIntegrity()
+        {
+            var oldState = _storageService.GetAsync<NSData>("biometricState").Result;
```

Comment:
shouldnt we just make this all async so that we don't have to deal with `Result`